### PR TITLE
Small mailserver async fetching refactor

### PR DESCRIPTION
**Pull request by vitvly**
Two small refactors extracted from https://github.com/status-im/status-go/pull/5292

### CODE REVIEW

**Review by osmaczko — APPROVED** [review]

File: protocol/messenger.go [hunk 1]
```diff
@@ -1503,18 +1503,14 @@ func (m *Messenger) watchConnectionChange() {
 		// Instead we will poll the connection status.
 		m.logger.Warn("using WakuV1, can't watch connection changes, this might be have side-effects")
 		go pollConnectionStatus()
-		return
-	}
+	} else {
```
Comment (igor-sirotin) [inline protocol/messenger.go]: I actually think `return` is better comparing to `else`. 
It's a check of `err != nil`, so returning makes sense there. If we will have to add more code with conditions after `go subscribedConnectionStatus(subscription)`, the tabulation will grow to the right.

File: protocol/messenger_mailserver_cycle.go [hunk 2]
```diff
@@ -434,14 +434,7 @@ func (m *Messenger) connectToMailserver(ms mailservers.Mailserver) error {
...
 			if m.config.codeControlFlags.AutoRequestHistoricMessages {
-				go func() {
-					_, err := m.performMailserverRequest(&ms, func(_ mailservers.Mailserver) (*MessengerResponse, error) {
-						return m.RequestAllHistoricMessages(false, false)
-					})
-					if err != nil {
-						m.logger.Error("could not perform mailserver request", zap.Error(err))
-					}
-				}()
+				m.asyncRequestAllHistoricMessages()
```
Comment (igor-sirotin) [inline protocol/messenger_mailserver_cycle.go:437]: `asyncRequestAllHistoricMessages` doesn't use `performMailserverRequest`.
Not sure if we want it, but noting the difference.

File: protocol/messenger.go (hunk 1 above)
Comment (vitvly) [reply protocol/messenger.go]: Agree, fixed.

File: protocol/messenger_mailserver_cycle.go (hunk 2 above)
Comment (vitvly) [reply protocol/messenger_mailserver_cycle.go:437]: Sorry, this one i don't quite understand - `asyncRequestAllHistoricMessages` does invoke `performMailserverRequest` here https://github.com/status-im/status-go/blob/e1e04e6f92be0515b6ada4a5143e4728e02991be/protocol/messenger_mailserver.go#L362

Comment (igor-sirotin) [reply protocol/messenger_mailserver_cycle.go:437]: oh, I got confused
So we were calling `performMailserverRequest` -> `RequestAllHistoricMessages` -> `performMailserverRequest` 😵‍💫

I guess then it's ok now

**Review by igor-sirotin — APPROVED** [review]

**Pull request merged**